src/compiler/parser.ts

```diff
@@ -10466,7 +10466,7 @@ function getNamedArgRegEx(name: string): RegExp {
 }
 
 const tripleSlashXMLCommentStartRegEx = /^\/\/\/\s*<(\S+)\s.*?\/>/im;
-const singleLinePragmaRegEx = /^\/\/\/?\s*@(\S+)\s*(.*)\s*$/im;
+const singleLinePragmaRegEx = /^\/\/\/?\s*@([^\s:]+)[\s:]*(.*)\s*$/im;

```

Comment:
If you write `@ts-check: : : : comment`, I think that the `: : : comment` should be the comment, not just `comment`.

I also think that colon should only be skipped right after ts-check. So something more like

```suggestion
const singleLinePragmaRegEx = /^\/\/\/?\s*@(\S+):?\s*(.*)\s*$/im;
```

Reply:
I've used a slightly different regex, beacuse `(\S+)` is greedy and will include any colons within that first match group

as per your example above, the comment segment is now `: : : comment`